src/ast/ast.c

```diff
@@ -430,6 +433,17 @@ const char **AST_BuildCallColumnNames(const cypher_astnode_t *call_clause) {
 	return proc_output_columns;
 }
 
+const char *AST_ExtractQueryString(const cypher_parse_result_t *partial_result) {
+	// Retrieve the AST root node from a parsed query.
+	const cypher_astnode_t *statement = cypher_parse_result_get_root(partial_result, 0);
+	// We are parsing with the CYPHER_PARSE_ONLY_STATEMENTS flag,
+	// and double-checking this in AST validations
```

Comment:
At this point, we've only parsed the input with `CYPHER_PARSE_ONLY_PARAMETERS`. I think this comment needs to more clearly describe that we're retrieving the query string from a partially-constructed AST.